addons/website_sale/controllers/main.py

```diff
@@ -561,11 +561,7 @@ def remove_product_image(self, image_res_model, image_res_id):
             raise NotFound()
 
         image_res_id = int(image_res_id)
-        if image_res_model == 'product.product':
-            request.env['product.product'].browse(image_res_id).write({'image_1920': False})
-        elif image_res_model == 'product.template':
-            request.env['product.template'].browse(image_res_id).write({'image_1920': False})
```

Comment:
Which case is handled by this code if we always submit a `'product.product'` or `'product.image'` field?

Reply:
Indeed, I do not see neither which case is handled here as, if I am not mistaken, it is impossible to directly remove a `product.template` (It is possible to modify it but not delete it (see picture))
![image](https://user-images.githubusercontent.com/114914971/224288372-210689fe-231b-42de-a9a8-71268a589e9d.png)